app/src/main/java/com/example/sharingang/items/FirestoreItemRepository.kt

```diff
@@ -0,0 +1,85 @@
+package com.example.sharingang.items
+
+import android.util.Log
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.google.firebase.firestore.Query
+import com.google.firebase.firestore.ktx.firestore
+import com.google.firebase.firestore.ktx.firestoreSettings
+import com.google.firebase.ktx.Firebase
+import kotlinx.coroutines.tasks.await
+import java.lang.Exception
+
+private const val TAG = "FirestoreItemRepository"
+
+class FirestoreItemRepository(useFirebaseEmulator: Boolean = false) : ItemRepository {
+    private val firestore = Firebase.firestore
+    private val collectionName = "items"
+
+    init {
+        if (useFirebaseEmulator) {
+            // 10.0.2.2 is the special IP address to connect to the 'localhost' of
+            // the host computer from an Android emulator.
+            firestore.useEmulator("10.0.2.2", 8080)
```

Comment:
If you are testing on a real phone, you might need to replace it with the IP of your computer